spec/controllers/admin_users_controller_spec.rb

```diff
@@ -0,0 +1,203 @@
+require 'rails_helper'
+
+RSpec.describe Admin::UsersController do
+  include Devise::Test::ControllerHelpers
+  include EmailSpec::Helpers
+  include EmailSpec::Matchers
+  render_views
+
+  describe '#index' do
+    context 'admin is sign in' do
+      it 'allows to search users with postgres fts' do
+        admin = create(:admin, first_name: 'Luke')
+        user = create(:user, first_name: 'Pablo', organization: admin.organization)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
+        sign_in admin
+
+        get :index, params: { search: { text: admin.first_name } }
+
+        expect(response.body).to include admin.first_name
+        expect(response.body).to_not include user.first_name
+      end
+    end
+
+    context 'admin is not sign in' do
+      it 'redirects user without admin privileges to home page' do
+        user = create(:user)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
+        sign_in user
+
+        get :index
+
+        expect(response).to redirect_to(root_path)
+      end
+    end
+  end
+
+  describe '#show' do
+    context 'admin is sign in' do
+      it 'redirects admin when accessing interest outside his organization' do
+        organization = create(:organization)
+        admin = create(:admin, organization: organization)
+        user = create(:user)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
+        sign_in admin
+
+        get :show, params: { id: user.id }
+
+        expect(response).to redirect_to(admin_users_path)
+      end
+    end
+
+    context 'admin is not sign in' do
+      it 'redirects user without admin privileges to home page' do
+        user = create(:user)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
+        sign_in user
+
+        get :show, params: { id: user.id }
+
+        expect(response).to redirect_to(root_path)
+      end
+    end
+  end
+
+  describe '#new' do
+    context 'admin is not sign in' do
+      it 'redirects user without admin privileges to home page' do
+        user = create(:user)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
+        sign_in user
+
+        get :new
+
+        expect(response).to redirect_to(root_path)
+      end
+    end
+  end
+
+  describe '#create' do
+    context 'admin is sign in' do
+      it 'rerender new page when empty user is provided' do
+        admin = create(:admin)
+        request.env['devise.mapping'] = Devise.mappings[:admin]
```

Comment:
`request.env['devise.mapping'] = Devise.mappings[:admin]` can be moved to a spec helper or a before block